src/reason-parser/reason_parser.mly

```diff
@@ -132,6 +132,7 @@ open Ast_mapper
 
 *)
 
+let bs_payload loc = ({loc; txt = "bs"}, PStr [])
```

Comment:
I think that this should be a general purpose ppx attribute called `@uncurried` which isn't specific to buckleScript - at least at parse time. Thoughts? Then in the reactjs jsx transform which is where most of the BS logic is located, we can turn these into `@bs` attributes.

Reply:
I actually started out by parsing an `@uncurried` attribute and transforming it to `@bs`.
One thing that kept me from going that route is existing code:
Imagine having
```reason
let f = [@bs] (a, b) => a + b;
```
One of the current "strengths" of our printer is the automatic transformation to
```reason
let f = (. a, b) => a + b;
```
Refmt teaches developers that `[@bs] (a, b) => ...` can be written in a much shorter way as `(. a, b) => ...`.

I'm actually in favour of parsing the `@uncurry` and delegating to `React ppx`, as a way to make Reason not Bs dsl, but the "teachable moment" kept it from here.
Am I missing something?
Should we just parse `@uncurry` and risk people typing `[@bs] (. a, b) => a + b`?
@chenglou What do you think?

Reply:
If printing were to be lossy, you could print either form with the leading dot. It would be reparsed as uncurried. That would provide the teachable moment, but at the expense of removing the desirable properties of refmt: that formattting is never lossy with respect to the AST. At least it would be idempotent.
Perhaps we could make an exception for lossinees this one time and then in the next version stop auto converting ‘@bs’ to ‘@uncurried’

Reply:
Sounds good, I'll get right to it.

Reply:
Discussed a little. This shouldn't be in the ReactJS JSX PPX as it has nothing to do with it. Not just because of the name, but because not everyone uses the ReactJS JSX PPX. And it'd be really confusing if folks want to have another JSX transform then suddenly have the @ uncurry -> @ bs logic change.

The clean way to solve it would be for BuckleScript to also support @ uncurry as an alias to @ bs. We shouldn't try to mingle here.

In the meantime, since no one uses @ bs for anything other than the BuckleScript uncurrying logic, I think we should keep it this way. Heck, touching @ uncurry might actually be bad right now by the same logic.

Reply:
"In the meantime, since no one uses @ bs for anything other than the BuckleScript "

That wasn't my concern. My concern was that we want to use the uncurry syntax outside of bucklescript. I don't think there should be hard coded assumptions on BuckleScript in the Reason parser. If you want to have this use "@bs" attributes it will likely break in the future. Right now, nothing else is using the uncurried syntax, but if it does, it must be okay to move Reason to use a different ppx attribute "@uncurried" (or something like that) without anyone getting upset that their BuckleScript "@bs" uncurrieds broke. If that's not okay, then the solution is to make it so that no one is unhappy by having Reason parser parse into some bs-agnostic attribute such as "@uncurried", and either have BS pay attention to "@uncurried", or apply some post-processing to turn it into "@bs".

Reply:
Are you saying people are using the dot notation while using another ppx that also looks for @ bs tag and transforms it in some custom way, while also not use bucklescript? Even if that's the case, it's easy for _those_ ppxes to handle both @ bs and @ uncurry